### PR TITLE
[1/Q] Add DB insert for volunteering opportunity

### PR DESCRIPTION
**Pull request by nikhitashanker**
UI for add volunteering opportunity:
image

After add row is clicked once:
image

- Add eventId field to volunteering opportunity
- Add UI to dynamically add new skill input fields
- Add Wrapper functions for insertion of volunteering opportunities
- Connect UI with DB
- Add a utilities to get parameters based on forms for both single value and multiple values

### CODE REVIEW

File: project/src/main/java/com/google/sps/data/VolunteeringOpportunity.java [hunk 1]
```diff
@@ -2,25 +2,29 @@
 
 import java.util.HashSet;
 import java.util.Set;
+import java.util.UUID;
 
 public final class VolunteeringOpportunity {
-  private final long opportunityId =
-    10000000L; // TODO: Needed for storing ID from DB, replace using ID generated by DB
+  private String opportunityId;
+  private String eventId;
```
Comment (dash102) [inline project/src/main/java/com/google/sps/data/VolunteeringOpportunity.java:9]: idea, maybe eventId can be replaced with the Event itself in order to lower the number of DB calls (so you don't have to query for other event fields and instead have it all in the object)

Comment (nikhitashanker) [reply project/src/main/java/com/google/sps/data/VolunteeringOpportunity.java:9]: yes, ill think about this - I am leaning towards using using an join between events and volunteering opportunities table.

File: project/src/main/webapp/create-volunteering-opportunity.html [hunk 2]
```diff
@@ -0,0 +1,67 @@
+<!DOCTYPE html>
```
Comment (liangqigong) [inline project/src/main/webapp/create-volunteering-opportunity.html:1]: Next time could you separate frontend change and backend change in different PRs?

Comment (dash102) [reply project/src/main/webapp/create-volunteering-opportunity.html:1]: Will do, sorry about that

Comment (nikhitashanker) [reply project/src/main/webapp/create-volunteering-opportunity.html:1]: Ack, will try to do this.

File: project/src/main/java/com/google/sps/servlets/VolunteeringFormHandlerServlet.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
+package com.google.sps.servlets;
```
[the remark this answers is not in the document]
Comment (nikhitashanker) [reply project/src/main/java/com/google/sps/servlets/VolunteeringFormHandlerServlet.java:1]: Done.

File: project/src/main/java/com/google/sps/utilities/CommonUtils.java [hunk 4]
```diff
@@ -1,9 +1,28 @@
 package com.google.sps.utilities;
```
[the remark this answers is not in the document]
Comment (nikhitashanker) [reply project/src/main/java/com/google/sps/utilities/CommonUtils.java:1]: Done.

**Comment by nikhitashanker** [comment]
Ready for review.

File: project/src/test/java/com/google/sps/CommonUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import org.apache.commons.lang3.StringUtils;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.springframework.mock.web.MockHttpServletRequest;
+
+/** */
```
Comment (liangqigong) [inline project/src/test/java/com/google/sps/CommonUtilsTest.java:11]: Remove

File: project/src/test/java/com/google/sps/CommonUtilsTest.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        StringUtils.EMPTY, CommonUtils.getParameter(request, NAME, StringUtils.EMPTY));
+  }
+
+  @Test
+  public void getParameterWithNonemptyValuesReturnValues() {
+    MockHttpServletRequest request = new MockHttpServletRequest();
+    request.setParameter(NAME, NONEMPTY_VALUES);
+
+    MatcherAssert.assertThat(
+        CommonUtils.getParameterValues(request, NAME), Matchers.contains(JOE, JAMES));
```
Comment (liangqigong) [inline project/src/test/java/com/google/sps/CommonUtilsTest.java:44]: When I read the test, I have no idea JOE and JAMES are in the NONEMPTY_VALUES until I read the constants. This is against the best practice.

https://docs.google.com/document/d/1M9kKw-Qgwmi-fhIDab-HCsK6sp53qRJQaM7aM3s4rXs/edit#heading=h.cxwr9nos46ee

I think you can remove the NONEMPTY_VALUES. It is true that we prefer to use constants in production codes, but it is not the same for testing codes.

File: project/src/test/java/com/google/sps/CommonUtilsTest.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  @Test
+  public void getParameterWithFewEmptyValuesReturnNonemptyValues() {
+    MockHttpServletRequest request = new MockHttpServletRequest();
+    request.setParameter(NAME, SOME_EMPTY_VALUES);
```
Comment (liangqigong) [inline project/src/test/java/com/google/sps/CommonUtilsTest.java:50]: ditto.

File: project/src/test/java/com/google/sps/CommonUtilsTest.java (hunk 5 above)
Comment (nikhitashanker) [reply project/src/test/java/com/google/sps/CommonUtilsTest.java:11]: Done.

File: project/src/test/java/com/google/sps/CommonUtilsTest.java (hunk 6 above)
Comment (nikhitashanker) [reply project/src/test/java/com/google/sps/CommonUtilsTest.java]: Done - I see, will keep this in mind.

File: project/src/test/java/com/google/sps/CommonUtilsTest.java (hunk 7 above)
Comment (nikhitashanker) [reply project/src/test/java/com/google/sps/CommonUtilsTest.java]: Done.

**Comment by nikhitashanker** [comment]
Ready for review.

**Pull request merged**